etc/dovetail-edge-deploy.yml

```diff
@@ -0,0 +1,865 @@
+    Type: String
+  ProductionBytesLambdaFunctionArn:
+    Description: >
+      e.g., arn:aws:lambda:<region>:<account>:function:<name>
+    Type: String
+  CertificateArn:
+    Type: String
+    Description: eg. arn:aws:acm:<region>:<account>:certificate/<guid>
+Resources:
+  # Pipeline
+  PipelineArtifactStore:
+    # The bucket used to store artifacts generated by CodePipeline actions.
+    # This bucket is to be used exclusively by AWS; don't use it to store any
+    # app, infrastructure, or user data.
+    Type: AWS::S3::Bucket
+    DeletionPolicy: Retain
+    Properties:
+      LifecycleConfiguration:
+        Rules:
+          - ExpirationInDays: 14
+            Status: Enabled
+      Tags:
+        - Key: Project
+          Value: Infrastructure
+        - Key: prx:cloudformation:stack-name
+          Value: !Ref AWS::StackName
+        - Key: prx:cloudformation:stack-id
+          Value: !Ref AWS::StackId
+  PipelineIamRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - codepipeline.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: "/"
+      Policies:
+        # CodePipeline seems to need access to special buckets
+        - PolicyName: GenericCodePipelineS3PutPolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:PutObject
+                Resource:
+                  - arn:aws:s3:::codepipeline*
+        # Let CodePipeline read and write artifacts
+        - PolicyName: ArtifactStoreFullAccess
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:Get*
+                  - s3:Put*
+                  - s3:List*
+                Resource:
+                  - !Sub arn:aws:s3:::${PipelineArtifactStore}
+                  - !Sub arn:aws:s3:::${PipelineArtifactStore}/*
+        # Allows CodePipeline to invoke Lambda function actions
+        - PolicyName: LambdaRolePolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "lambda:ListFunctions"
+                Resource: "*"
+              - Effect: Allow
+                Action:
+                  - "lambda:InvokeFunction"
+                Resource:
+                  - !GetAtt DiscoverEdgeLambdaVersionsFunction.Arn
+  Pipeline:
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      ArtifactStore:
+        Location: !Ref PipelineArtifactStore
+        Type: S3
+      RoleArn: !GetAtt PipelineIamRole.Arn
+      Stages:
+        - Name: Source
+          Actions:
+            - Name: InfrastructureRepo
```

Comment:
The Infrastructure repo is an input to this?  That seems odd - isn't this edge-deploy stack manual?

Reply:
It needs to be an input so we can get the YAML file to pass to Cfn to deploy